packages/vite-extensions/vite-list-assets-required-for-offline-mode.ts

```diff
@@ -0,0 +1,112 @@
+import { readdirSync, statSync, writeFileSync } from 'node:fs';
+import { join } from 'node:path';
+
+/**
+ * We want to cache all files from the website root directory, except for the files listed below.
+ */
+const patternsToNotCache = [
+	/**
+	 * Static files that are not needed for the website to function offline.
+	 */
+	'/package.json',
+	'/README.md',
+	'/.DS_Store',
+	'/index.cjs',
+	'/index.d.ts',
+	/\/lib\/.*/, // Remote lib files
+	/\/test-fixtures\/.*/, // Test fixtures
+	/**
+	 * WordPress assets removed from the minified builds, for example:
+	 *
+	 *      /wp-6.2/wp-content/themes/twentytwentyone/style.css
+	 *
+	 * We don't need to cache them, because they're only used in a short time
+	 * window after the Playground is initially loaded and before they're all backfilled
+	 * from a dedicated ZIP file shipping all the static assets for a given minified
+	 * build.
+	 *
+	 * See <Assets backfilling PR link>
+	 */
+	/^\/wp-[\w+((\.\d+)?)]+\/.*/,
+	/**
+	 * Demos are not needed for the website to function offline.
+	 */
+	/^\/demos\/.*/,
+	/**
+	 * Files needed only by the Playground.WordPress.Net server.
+	 */
+	'/.htaccess',
+	/**
+	 * We can't download the PHP scripts – only get their execution result. This is fine,
+	 * we don't need them for the offline mode anyway. This includes things like plugins-proxy.php.
+	 */
+	/\/.*\.php$/,
+	/**
+	 * WordPress, PHP, and SQLite files that are loaded during boot.
+	 * By loading these files during boot, we can ensure that only the current PHP/WordPress/SQLite files are cached.
+	 * These files are large, so loading only the current files will reduce the bandwidth usage.
```

Comment:
Let's expand this a bit for clarity:

```suggestion
	 * Eagerly loading all the PHP and WordPress releases offered by Playground would use an
	 * extra ~200MB of bandwidth every time you load Playground on a new device.
	 *
	 * However, most of the time time you only want to load a specific Playground configuration.
	 *
	 * Therefore, in here we're excluding the PHP and WP releases from being loaded
	 * eagerly and instead we're defaulting to caching the specific release that's 
	 * loaded anyway when booting Playgroung.
```